cmd/kubeadm/app/cmd/join.go

```diff
@@ -129,40 +128,17 @@ func (j *Join) Validate() error {
 	return validation.ValidateNodeConfiguration(j.cfg).ToAggregate()
 }
 
-// Run executes worked node provisioning and tries to join an existing cluster.
+// Run executes worker node provisioning and tries to join an existing cluster.
 func (j *Join) Run(out io.Writer) error {
-	var cfg *clientcmdapi.Config
-	// TODO: delete this first block when we move Token to the discovery interface
-	if j.cfg.Discovery.Token != nil {
-		clusterInfo, err := kubenode.RetrieveTrustedClusterInfo(j.cfg.Discovery.Token)
-		if err != nil {
-			return err
-		}
-
-		connectionDetails, err := kubenode.EstablishMasterConnection(j.cfg.Discovery.Token, clusterInfo)
-		if err != nil {
-			return err
-		}
-		err = kubenode.CheckForNodeNameDuplicates(connectionDetails)
-		if err != nil {
-			return err
-		}
-		cfg, err = kubenode.PerformTLSBootstrapDeprecated(connectionDetails)
-		if err != nil {
-			return err
-		}
-	} else {
-		cfg, err := discovery.For(j.cfg.Discovery)
-		if err != nil {
-			return err
-		}
-		if err := kubenode.PerformTLSBootstrap(cfg); err != nil {
-			return err
-		}
+	cfg, err := discovery.For(j.cfg.Discovery)
+	if err != nil {
+		return err
+	}
+	if err := kubenode.PerformTLSBootstrap(cfg); err != nil {
```

Comment:
Yup, we need to move kube-discovery over to using a kubeconfig rather than clusterinfo in order to move it behind the Discovery interface. Might be too much for this PR :) cc @dgoodwin